3.0.0 (#1)
mu-ts/logger: README.md

### @mu-ts/logger
# Simple lightweight logging.

> This logging frameowrk aims to be a good enough logging framework where being lightweight is an asset. For example FaaS situations like AWS Lambda, Azure Functions, , Google Functions, etc. There are a lot out there. A common problem is cold start, which bundle size has an impact.

Reviewer: ```suggestion
This logging framework aims to be a good enough logging framework where being lightweight is an asset. For example FaaS situations like AWS Lambda, Azure Functions, Google Functions, etc. There are a lot out there. A common problem is cold start, where bundle size has an impact.
```

##### Synchronous Example

```
{
  at: 2019-12-07T20:17:41.992Z,
  clazz: 'ClassX',
  func: 'doSync()',
  msg: 'inOut -->',
  data: { args: [ 'Hello Viewer' ] },
  name: 'ClassX.inOut',
  level: 'debug'
}
{
  at: 2019-12-07T20:17:42.005Z,
  clazz: 'ClassX',
  func: 'doSync()',
  msg: 'inOut <--',
  data: { result: 'x' },
  name: 'ClassX.inOut',
  level: 'debug'
}
```

##### Asynchronous Functions

In the case of async functions, you will see 3 output statements as an additional one is added to record the fact the method returned a Promise, but did not yet resolve.

```
{
  at: 2019-12-07T20:17:42.007Z,
  clazz: 'ClassX',
  func: 'doASync()',
  msg: 'inOut -->',
  data: { args: [ 'Hello Longer Guy' ] },
  name: 'ClassX.inOut',
  level: 'debug'
}
{
  at: 2019-12-07T20:17:42.013Z,
  clazz: 'ClassX',
  func: 'doASync()',
  msg: 'inOut -- promisified function',
  data: { result: {} },
  name: 'ClassX.inOut',
  level: 'debug'
}
{
  at: 2019-12-07T20:17:42.016Z,
  clazz: 'ClassX',
  func: 'doASync()',
  msg: 'inOut <--',
  data: { result: 'x' },
  name: 'ClassX.inOut',
  level: 'debug'
}
```

### @duration({level, name})

> So this decorator will tell you the amount of time it takes for your function to execute. If this is an async function, it will be the amount of time util the promise resolves, not the amount of time to return the promise.

Reviewer: ```suggestion
This decorator will tell you the amount of time it takes for your function to execute. If this is an async function, it will be the amount of time until the promise resolves, not the amount of time to return the promise.
```

Default implementation outputs values using `console.timeEnd()` the name is assembled together using the logger name, 'duration' and function name.

Example: `X.duration.doASync: 0.366ms`

> But wait, thats not JSON! Yea, err'ing on the side of using out of the box behavior instead of build this out. Its easy to create memory leaks with this kind of behavior. When I do it, I want to take the time to do it right.

Reviewer: ```suggestion
But wait, that's not JSON! Yea, erring on the side of using out of the box behavior instead of build this out. It's easy to create memory leaks with this kind of behavior. When I do it, I want to take the time to do it right.
```

> ## Level's

Reviewer: ```suggestion
## Levels
```

> Pretty standard here, highest level is fatal, lowest (most detailed) is trace. Lowest to highest they are all listed below.

Reviewer: ```suggestion
Pretty standard here, highest level is fatal, lowest (most detailed) is trace. Levels listed below from lowest to highest:
```

* trace `isTrace()` `trace(...params)`
* debug `isDebug()` `debug(...params)`
* info `isInfo()` `info(...params)`
* warn `isWarn()` `warn(...params)`
* error `isError()` `error(...params)`
* fatal `isFatal()` `fatal(...params)`

> For the default logger, each of tehse levels will use a different output method to take advantage of some of the built in functionality of `console`. So trace, debug and info are all reported using `console.log`, warn is reported using `console.warn` and then error and fatal are reported using `console.error`.

Reviewer: ```suggestion
For the default logger, each of these levels will use a different output method to take advantage of some of the built in functionality of `console`. So trace, debug and info are all reported using `console.log`, warn is reported using `console.warn` and then error and fatal are reported using `console.error`.
```
